src/main/java/com/esri/samples/editing/edit_and_sync_features/EditAndSyncFeaturesSample.java

```diff
@@ -156,41 +198,47 @@ public void start(Stage stage) {
 
           // iterate through all feature layers in the map
           for (Layer layer : mapView.getMap().getOperationalLayers()) {
-            final FeatureLayer featureLayer = (FeatureLayer) layer;
+            FeatureLayer featureLayer = (FeatureLayer) layer;
 
-            // identify any clicked feature
+            // start a process to identify any features at the clicked screen point
             ListenableFuture<IdentifyLayerResult> results = mapView.identifyLayerAsync(featureLayer, screenPoint, 1, false, 1);
             results.addDoneListener(() -> {
               try {
-                // get selected feature, if it is a point
+                // get the results of the identification process to determine whether a feature was clicked on
                 List<GeoElement> elementList = results.get().getElements();
+                // determine whether the clicked feature represents a point that can be moved
                 if (elementList.size() > 0 && elementList.get(0) instanceof ArcGISFeature && elementList.get(0).getGeometry().getGeometryType() == GeometryType.POINT) {
 
-                  // clicked on a feature, select it
+                  // clear the previous selection (to avoid selecting multiple features at the same time)
+                  featureLayer.clearSelection();
+
+                  // grab the new feature and make it selected
                   ArcGISFeature selectedFeature = (ArcGISFeature) elementList.get(0);
-                  // TODO: this doesn't clear the selection in all FL, hence some elements stay selected?
-                  featureLayer.clearSelection(); // clear previous selection
                   featureLayer.selectFeature(selectedFeature);
 
                   // change the state to editing
                   currentEditState = EditState.EDITING;
 
                 } else {
-                  // didn't click on a feature
+                  // the user did not click on a feature (implying no selection was consciously made)
```

Comment:
Still confused by this comment. At this point, the user has clicked a feature and wants to move it: so the selection is still active at this point and a selection was consciously made?

Reply:
If the user has previously clicked on a feature and selected it, then we can assume the intention with a click on the map is to move it. However there's still the possibility that we don't have a feature selected at this point, and the user is simply clicking around the map. I made a diagram to hopefully make this section of the code a bit clearer. Happy to take suggestions on how to structure this differently (nested if loops don't really convey the intentions too well here)

![Untitled Diagram (1)](https://user-images.githubusercontent.com/44193148/57858860-b89ea200-77e9-11e9-9966-f2d47b6a8f1b.png)

Reply:
ok, in that case, I'd remove the first comment and leave only the "retrieve the selected features...." comment, IMO it's clearer that way.